### PR TITLE
18 elog click

### PR DESCRIPTION
**Pull request by pontuslaestadius**

### CODE REVIEW

File: Application/src/visuals/Draw.java [hunk 1]
```diff
@@ -226,4 +226,25 @@ public static void animate(boolean active) {
...
+     * @return a message.
+     */
+    public Message getMessage(int index) throws ArrayIndexOutOfBoundsException {
+        if (index >= this.messages.size())
+            throw new ArrayIndexOutOfBoundsException("Index: " + index + "Size: " + this.messages.size());
+        return this.messages.get(index);
+    }
+
+    /**
+     * Resets all the static to be current not static time line. It's the darkest time line.
```
Comment (IsabelleTornqvist) [inline Application/src/visuals/Draw.java]: darkest time line?

File: Application/src/visuals/ExecutionLog.java [hunk 2]
```diff
@@ -49,6 +57,66 @@ public ExecutionLog() {
         BackgroundFill bgf = new BackgroundFill(c, null,null);
         Background bg = new Background(bgf);
         this.listView.setBackground(bg);
+
+        // Adds listener for selecting a item.
+        this.listView.getSelectionModel().selectedItemProperty().addListener
+                ((ObservableValue<? extends String> observable, String oldValue, String newValue) -> {
```
Comment (IsabelleTornqvist) [inline Application/src/visuals/ExecutionLog.java]: Don't know what this means, can't give feedback on this line

File: Application/src/visuals/ExecutionLog.java [hunk 3]
```diff
@@ -1,17 +1,25 @@
 package visuals;
 
+import com.sun.xml.internal.ws.policy.privateutil.PolicyUtils;
```
Comment (ghost) [inline Application/src/visuals/ExecutionLog.java]: what does this do?

**Review by ghost — CHANGES_REQUESTED** [review]
shouldn't the logic handle resetting the offset when you click the back button to the previous step?

File: Application/src/visuals/ExecutionLog.java [hunk 4]
```diff
@@ -1,17 +1,25 @@
...
+import javafx.beans.value.ChangeListener;
```
Comment (ghost) [inline Application/src/visuals/ExecutionLog.java]: unused

File: Application/src/visuals/ExecutionLog.java [hunk 5]
```diff
@@ -1,17 +1,25 @@
...
+import javafx.beans.value.ObservableValue;
 import javafx.collections.FXCollections;
 import javafx.collections.ObservableList;
 import javafx.scene.control.*;
 import javafx.scene.layout.Background;
 import javafx.scene.layout.BackgroundFill;
 import javafx.scene.paint.Color;
 import javafx.scene.control.ListView;
+import visuals.handlers.Resizer;
```
Comment (ghost) [inline Application/src/visuals/ExecutionLog.java]: unused

**Review by ghost — APPROVED** [review]

**Review by timjon — CHANGES_REQUESTED** [review]
I cant click on the latest message to replay it

**Comment by pontuslaestadius** [comment]
@timjon That's intentional

**Review by kosara-g — COMMENTED** [review]
Sick dragon bro speed. Nice work overall. I have a few questions: why does it show a black border around the path of the dragon bro sometimes and how do I stop it from looping the message I'm replaying? Other than that, seems great.

**Review by kosara-g — APPROVED** [review]

**Review by IsabelleTornqvist — APPROVED** [review]

**Review by timjon — CHANGES_REQUESTED** [review]
I found 4 bugs in the 30 seconds I checked the feature so im going to block this

**Review by timjon — CHANGES_REQUESTED** [review]
1. the previous button replays the message before the one you removed. how do reproduce:
- Import a diagram with at least 4 messages
- Play through the first 3 messages
- Click on the previous button
- The dragon for message number 2 will now fly again
2. Clicking on a message in the execution log and then the last message in the execution log will make the next button animate the same message twice. 
How to reproduce:
- Import a diagram with at least 3 messages
- Simulate the first two messages
- Click on the first message to animate it again
- Click on the second message to animate it again
- Click on the next message button
- The dragon will now fly twice

**Review by ghost — CHANGES_REQUESTED** [review]
Automation button doesn't work after you latest commit

**Comment by kosara-g** [comment]
The animation stopped after 1 message for me and then none of the buttons worked but it didn't crash either.

**Review by timjon — APPROVED** [review]

**Review by ghost — APPROVED** [review]

**Review by Sebastian-Fransson — APPROVED** [review]
Auto seems to work now

**Pull request merged**